### PR TITLE
Update and clean-up compsets for use in test suites

### PR DESCRIPTION
**Pull request by wlin7**
Remove obsolete eam and allactive compsets and the associated eam
use_case files, and update the compsets used in nightly test suites.

All compsets referring to AV1C or v0atm are removed.  F20TRC5-CMIP6
and F2010SC5-CMIP6 are no more needed (previously retained for ongoing
testings).  Remaining use case files referring to av1c are removed, including  
those used by MMF tests. Also Remove obsolete use_case files referring to 
cam3 and cam4 .
This PR also makes all standard F-cases use prescribed data ice mode of MPASSI.
This includes F1850, F1950, F2010, F20TR. The corresponding compsets that
use CICE are renamed as F1850-CICE, F1950-CICE, F2010-CICE, and
F20TR-CICE, respecticely. The F####-CICE compsets set to go
with SROF while the MPASSI counterparts with MOSART as river component.

The following compsets are replaced in nightly test suites.

. FC5AV1C-L with F2010
. A_WCYCL1850 and A_WCYCL1850S_CMIP6 with WCYCL1850
. FC5AV1C-H01A with F2010 as there is no v2 compset just for hires
. A_WCYCL1950S_CMIP6_HR and A_WCYCL2000_H01AS with WCYCL1950 as a placeholder

With compset changes and the use of MPASSI data ice mode, F2010 tests using ne4_ne4 
change to use ne4_oQU240, and ne4pg2_ne4pg2 to ne4pg2_oQU480;
F2010 tests with ERS_Ln# and ERP_Ln# change to use ERS_Ld3.

[non-BFB] Virtually for all tests involving the compsets above. Tests using the old compsets would appear missing
          while new tests would appear correspondingly.

Note: The PR message is updated (new text in the 2nd and 3rd paragraph)

### CODE REVIEW

File: cime_config/allactive/config_compsets.xml
Comment (rljacob) [inline cime_config/allactive/config_compsets.xml:275]: Can we rename this one to WCYCL1850-DIB  ?

File: cime_config/allactive/config_compsets.xml
Comment (rljacob) [inline cime_config/allactive/config_compsets.xml:303]: Lets remove the "A_" from this one.

**Review by rljacob — APPROVED** [review]

**Comment by rljacob** [comment]
What we should do to verify bfb after the name changes is run cprnc by hand for a few cases on Chrysalis.

File: cime_config/allactive/config_compsets.xml
Comment (jonbob) [inline cime_config/allactive/config_compsets.xml:420]: We may want to replace this one with something else

File: cime_config/allactive/config_compsets.xml
Comment (wlin7) [reply cime_config/allactive/config_compsets.xml:275]: @jonbob , can we remove this one? Look like CRYO1850 is the new equivalent?

**Comment by wlin7** [comment]
> What we should do to verify bfb after the name changes is run cprnc by hand for a few cases on Chrysalis.

Good idea. How long the baselines for the would-be missing tests be saved? I would certainly do cprnc before merging to master and requests for new baselines.

File: cime_config/allactive/config_compsets.xml
Comment (jonbob) [reply cime_config/allactive/config_compsets.xml:420]: How about making it BGWCYCL1850_CMIP6 and changing "EAM%AV1C" to "EAM%CMIP6"?

File: cime_config/allactive/config_compsets.xml
Comment (jonbob) [reply cime_config/allactive/config_compsets.xml:275]: @wlin7 - which one do you want to remove?

Comment (jonbob) [reply cime_config/allactive/config_compsets.xml:275]: I'm asking the cryo people about the older compsets -- the A_WCYCL1850-DIB-ISMF_CMIP6 is almost exactly the same as the new CRYO1850 (only different in specifying spunup ocn/ice IC's)

**Comment by wlin7** [comment]
With renamed F-compsets following the commit of 864ad0c, it has been verified that the new F2010 and F20TR are BFB with previous F2010-MPASSI and F20TR-MPASSI, respectively. Similarly, the new F2010-CICE and F20TR-CICE are BFB with previous F2010 and F20TR, respectively.

File: cime_config/allactive/config_compsets.xml
Comment (wlin7) [reply cime_config/allactive/config_compsets.xml]: A_WCYCL1850-DIB_CMIP6 and A_WCYCL1850-DIB_CMIP6 are now removed.

File: cime_config/allactive/config_compsets.xml
Comment (wlin7) [reply cime_config/allactive/config_compsets.xml:420]: A_BGWCYCL1850 has been removed. BGC compsets for v2 are yet to be finalized.

File: cime_config/allactive/config_compsets.xml
Comment (wlin7) [reply cime_config/allactive/config_compsets.xml:293]: Done.

**Comment by jonbob** [comment]
@wlin7 - the A_BGWCYCL1850 compset was just added and should not be removed. It's a cryo compset and not BGC (though it's hard to tell from the naming). Can it be added back, but without the A_ prefix?

**Comment by wlin7** [comment]
> @wlin7 - the A_BGWCYCL1850 compset was just added and should not be removed. It's a cryo compset and not BGC 

Thanks, @jonbob . I didn't realize it was recently added. I removed it because it was using AV1C. I am going to change it to EAM%CMIP6 as you suggested.

**Review by jonbob — APPROVED** [review]
Approved based on visual inspection

**Comment by wlin7** [comment]
Merged to next. 
Conflict in cime_config/allactive/config_compsets.xml due to new CRYO1950. Conflicts in components/eam/cime_config/config_compsets.xml due to naming changes among F###, F###-CICE and F###-MPASSI (some of which introduced in #4483).

**Comment by wlin7** [comment]
Reverted due to missing input files for F2010 to go with test grids (mostly ne4_ne4).

**Comment by wlin7** [comment]
No suitable mpassi mesh and graph file available to work with ne4np4 (nor ne4pg2) grid that is broadly used for F-case tests.

**Comment by jonbob** [comment]
I think these will have to change the resolution to be ne4_oQU480 or ne4_oQU240 -- mpassi will have to run on an mpas grid

**Comment by rljacob** [comment]
That's a good change because it makes the F-cases more resemble the coupled cases.   Traditionally, the all-one-resolution F-cases were for CAM to run with its own driver and no coupler.

**Comment by wlin7** [comment]
@rljacob ,  Jon and I also iterated on this grid issue on slack. We decided to replace all ne4_ne4 with ne4_oQU240 and ne4pg2_ne4pg2 with ne4pg2_oQU480 (no ne4pg2_oQU240 available).

There is another issue with coupled case. The standard WCYCL1850 sets to always use spun-up ocean/ice. Such spun-up ICs not available for small grids of oQU240/oQU480. We would like to temporarily maintain a slightly different compset WCYCL1850T to go without spun-up IC. Inline comments will be included to note the compset is for the purpose of testing at small grids. Jon will create ICs for the small grids and once ready,  putting in a PR to replace WCYCL1850T. Does this sound ok with you

**Comment by rljacob** [comment]
Yes that's fine.

I notice a couple of compsets still have "A_" in them:
A_WCYCL1850_AR5emis_superfast  and A_WCYCL20TR_AR5emis_superfast.   Should those be changed?

**Comment by rljacob** [comment]
Another option for the no-spinup is to call it  WCYCL1850NS and the long name would be
1850NS_EAM%CMIP6_ELM%SPBC_MPASSI_MPASO_MOSART_SGLC_SWAV

You could maybe leave it that way since we don't really need spun-up oceans for those tests do we?

**Comment by jonbob** [comment]
No, I think we figured making spunup initial conditions for those test grids made them sound scientifically supported somehow. The longname for WCYCL1850NS wouldn't even have to have 1850NS_... -- it could be 1850_ or really anything except 1850SOI_

**Comment by rljacob** [comment]
Sure just "1850_" is fine.

**Comment by wlin7** [comment]
Sounds great. I am going with WCYCL1850NS for alias, and just 1850_* for long name, Using 1850NS_* would trigger some additional changes in cime (in config_component files).

File: cime_config/tests.py
Comment (wlin7) [inline cime_config/tests.py:404]: For this set of tests, not all the grids are ready for F2010 with mpassi for prescribed ice. Only ne4_ne4 that is also replaced for the integration tests is using the new grid to go with F2010. F2010-CICE is used for the other grids, as these tests are just for eam_nl.

**Comment by wlin7** [comment]
Verified BFB tests from previous merge:
```
SMS_D_Ld1.ne30pg2_r05_EC30to60E2r2.WCYCL1850 (new) --> SMS_D_Ld1.ne30pg2_r05_EC30to60E2r2.A_WCYCL1850S_CMIP6
SMS_Ld1.ne30pg2_r05_EC30to60E2r2.WCYCL1850.chrysalis_intel.allactive-wcprod --> SMS_Ld1.ne30pg2_r05_EC30to60E2r2.A_WCYCL1850S_CMIP6.chrysalis_intel.allactive-wcprod
SMS_PS.northamericax4v1pg2_WC14to60E2r3.WCYCL1850.chrysalis_intel.allactive-wcprodrrm --> SMS_PS.northamericax4v1pg2_WC14to60E2r3.A_WCYCL1850S_CMIP6.chrysalis_intel.allactive-wcprodrrm

```
These tests won't be affected after this PR is  remerged to next. One wcprod test has been using WCYCL1850 previously. That test passed.

**Comment by rljacob** [comment]
Did you run e3sm_developer?

**Comment by wlin7** [comment]
@rljacob , I ran the developer test suite. Some ERS tests failed because of the run length and time step for the new ROF (MOSART) unable to synchronize the components for a same restart points.  I am testing extending the run length with the _L# option. It also appears ERS_Ln# option would not work.

**Comment by rljacob** [comment]
Maybe switching all F-cases to MPASSI should wait for another PR.  What we really want from this PR is the compset alias name changes.

**Comment by wlin7** [comment]
@rljacob . This is now fixed with ERS_Ld3 option (a120bae).  It should be better we have the MPASSI change in this PR, or we would  another around of as many diff.

Some explanation on using the ERS_Ld3 option (or with _L option) at all for the new F2010 tests.  _Ln# option wouldn't work because ROF is always ahead of the other components, and would have written the 2nd restart file if the run length is just 2*REST_N-1, as by design for the _Ln# option.

Take the example of ERS_Ln11 between atm and rof.  atm would write restart after STEP 6, ROF write restart at the time corresponding to atm's STEP 6, but would also write the 2nd restart corresponding to atm's STEP 12, which atm would have partially completed by the end of the run.

**Comment by wlin7** [comment]
Reverted the previous revert of this merging this PR, then remerged the branch to next.
No conflicts on remerging for cime_config/allactive/config_compsets.xml or components/eam/cime_config/config_compsets.xml

conflicts in components/eam/cime_config/config_component.xml due to changes in compset matching pattern (additional character as in '[1XO]') for the use cases scam_arm97_MMF-1mom and scam_rico_MMF-1mom. Resolved to what are in the current master.

**Comment by wlin7** [comment]
Tests return:
Chrysalis DIFFs expected  and FAILs are accounted for.
ERS_Ln9.ne4pg2_ne4pg2.F-MMF1-RCEMIP.chrysalis_intel due to having use_htefrz_classnuc = .true.

**Comment by wlin7** [comment]
Merged to master. 
Requests submitted separately for those requiring blessing (MMF tests), and those needing new baselines.

F2010 and F20TR wcprod tests on anvil failed. These tests passed on chrysalis and compy.

**Comment by wlin7** [comment]
ERS_Ln9.ne4pg2_ne4pg2.F-MMF1-RCEMIP.chrysalis_intel passed (not a DIFF) after fixing compset matching for the use_case (via #4535).  conusx4v1 test becomes a DIFF after #4535.

ERP_Ld3.ne4_oQU240.F2010 returned an expected diff (new test) on compy, but  continued to fail on mappy for a different and unclear reason. The mappy test appears to have completed the run and trying to compare with non-existent baseline.

**Comment by wlin7** [comment]
wcprod tests with F2010 and F20TR continued to fail on anvil (while no problem on other machines).

chrysalis atm_nbfb_next: after having graph2880 file, still has two FAILs below -- MVK_PS test passed but still needs to be blessed (request submitted).
```
PGN_P1x1.ne4_oQU240.F2010.chrysalis_intel
TSC.ne4_oQU240.F2010.chrysalis_intel
```

**Comment by wlin7** [comment]
ERP_Ld3.ne4_oQU240.F2010.mappy_gnu was blessed anyway on Sept. 14. But the test continues being reported as a FAIL today, despite no apparent error or failure in the test log.

**Comment by rljacob** [comment]
@jgfouca can you figure out why ERP_Ld3.ne4_oQU240.F2010.mappy_gnu  is flagged as "FAIL" ?

**Comment by rljacob** [comment]
This ERP test is PASS on: chrysalis, anvil, compy
FAIL on: mappy, cori-knl, cori-haswell
the fails on cori are without baselines so something in the test itself is triggering "FAIL".  The only thing I see is memleak but I thought those were ignored.

**Comment by wlin7** [comment]
Request submitted to bless anvil wcprod F2010 and F20TR tests,  after the new default pelayout (#4541) fixed the runtime issue.

**Comment by wlin7** [comment]
The tests below failed after this PR. Theta also did these tests.

```
SMS.ne120_oRRS18v3.WCYCL1950.cori-knl_intel.eam-cosplite
SMS.ne120_ne120.F2010.cori-knl_intel 
```

**Comment by wlin7** [comment]
After updated to latest cime master with ESMCI's [#4092](https://github.com/ESMCI/cime/pull/4092), rhe following atm_nbfb tests executed and returned DIFF, though aborted during baseline comparison. Request submitted to bless the diffs

```
PGN_P1x1.ne4_oQU240.F2010.chrysalis_intel
TSC.ne4_oQU240.F2010.chrysalis_intel
```

**Comment by rljacob** [comment]
I alread did that in https://acme-climate.atlassian.net/browse/SES-1814

**Pull request merged**
